src/others/minkowski-distance.js

```diff
@@ -0,0 +1,79 @@
+(function (exports) {
+  'use strict';
+
+  var minkowskiDistance = (function () {
+
+    function chebyshevDistance (x, y, lx, p, mathfn) {
+      var ret = -p;
+      var i;
+
+      for (i = 0; i < lx; i += 1) {
+        ret = mathfn(ret, Math.abs(x[i] - y[i]));
+      }
+
+      return ret;
+    }
+
+    function minkowskiDistance (x, lx, y, ly, p) {
+      var d;
+      var i;
+
+      if (lx !== ly) {
+        throw '2 points must have same array length';
+      }
+
+      if (isNaN(p)) {
+        throw 'p must be a number';
+      }
+
+      if (p === Number.POSITIVE_INFINITY) {
+        return chebyshevDistance(x, y, lx, p, Math.max);
+      } else if (p === Number.NEGATIVE_INFINITY) {
+        return chebyshevDistance(x, y, lx, p, Math.min);
+      } else if (p < 1) {
+        throw 'p less than 1 will violate triangle inequality';
```

Comment:
Would you change it to: "Order less than 1 will violate the triangle inequality"?